cmd/collector/collector.go

```diff
@@ -173,13 +181,42 @@ func run() error {
 		}
 	}()
 
+	// Create a new context with a cancel function
+	_, cancel := context.WithCancel(context.Background())
+	var server *http.Server
+	// Start the HTTP server in a separate goroutine
+	go func() {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/record", flowRecordHandler)
+		mux.HandleFunc("/reset", resetRecordHandler)
+
+		server = &http.Server{
```

Comment:
you should just set `ReadHeaderTimeout` to avoid the gosec warning

Reply:
Fixed. Thanks.